docs/spec/index.md

```diff
@@ -1,54 +1,67 @@
+| stop_times_supplement.txt | Supplement | Supplements and modifies GTFS [stop_times.txt](https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md#stop_timestxt) with non-public times that trips stop at locations, and related information. |
+| routes.txt | Supplement | Supplements and modifies GTFS [routes.txt](https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md#routestxt) with internal route identifiers and other non-public route identification. |
+| runs_pieces.txt | TODS-Specific | Defines daily personnel schedules within a feed. |
+| run_events.txt | TODS-Specific| Defines other scheduled activities to be performed by a member of personnel during a run. |
 
-### deadhead_times.txt
+The use of the `_supplement` standard to modify other GTFS files remains experimental at this time.
+
+
+## `_supplement` Files
+
+### Structure & Evaluation
+
+The fields in all `_supplement` files match those defined in the corresponding file's [GTFS specification](https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md).
+
+Each row in a `_supplement` file is used to modify the corresponding file in GTFS and should be evaluated as follows:
+
+1. If the row's Primary Key is defined in the corresponding GTFS file, and the `TODS_delete` field is equal to `1`, remove the row from the GTFS file.
+2. If the row's Primary Key is defined in the corresponding GTFS file, and the `TODS_delete` field is not defined or not equal to `1`, set or update (in the corresponding GTFS file) the value of any non-blank fields defined in the `_supplement` file's row to the `_supplement`'s value.
+3. If the row's Primary key is NOT defined in the corresponding GTFS file, add the row to the corresponding GTFS file.
+
+In other words, where a primary key matches, the row is either removed or any non-empty values in the row are used to *update* the corresponding GTFS values. Where a primary key match does not exist, the entire row is added.
+
+> _**Implications and Guidance:**_
+> - To explicitely blank a field, delete the row entirely using the `TODS_delete` field and then re-add the row with the desired values subsequent thereto.
+> - If a row contains defined values besides the Primary Key and a `TODS_delete` value of `1`, the row shall be removed and other values in that row will be ignored.
+
+### TODS-Specific Fields
+
+In addition to the fields defined in GTFS, specific fields for use within TODS are denoted by a `TODS_` field prefix.
+
+| **File** | **Field Name** | **Type** | **Required** | **Description** |
+| --- | --- | --- | --- | --- |
+| Any `_supplement` file | `TODS_delete` | Enum | Optional | (blank) - Update other fields; do not delete.<br>`1` - Deletes the GTFS row in the corresponding file whose primary key matches the value in the `_supplement` file's row. |
+| `trips_supplement.txt` | `TODS_trip_type` | Text | Optional | Defines the type of the trip if distinct from a standard revenue trip. |
```

Comment:
What values would go here? Is this for deadheads? It might be nice to have a `is_in_passenger_service` boolean field that doesn't require matching against arbitrary text. That could replace this or be in addition to it.

Reply:
I like that concept! Thoughts on that terminology or something shorter like `TODS_revenue`?

I realize some operations may be fare-free, but the term "revenue service" is fairly well-defined in the US market, particularly given the FTA definition:

> Revenue Service (Miles, Hours, and Trips)
> The time when a vehicle is available to the general public and there is an expectation of carrying passengers. These passengers either: 
> •   Directly pay fares;
> •   Are subsidized by public policy; or
> •   Provide payment through some contractual arrangement.
>  
> Vehicles operated in fare free service are considered in revenue service. Revenue service includes: 
> •   Layover / recovery time.
>  
> Revenue service excludes: 
> •   Deadhead;
> •   Vehicle maintenance testing;
> •   School bus service; and
> •   Charter service.
> Can be found in: A-10, A-30, S-10, R-20, FFA-10, Declarations

Reply:
As I've tried modifying things for this, I am not sure if we should include this. Some vendors may which to use this boolean methodology to distinguish revenue trips from nonrevenue trips, but I think the better approach is letting each agency specify which values should be considered as revenue or nonrevenue service.

Perhaps that becomes a future TODS-specific file of `trip_types.txt` that maps specific values of the `TODS_trip_type` field to a revenue/nonrevenue or other related categorization that we can open as a new problem.

Reply:
One possible reading of the definition as-written is that if `TODS_trip_type` is null, the supplemental trip should be treated as a standard revenue trip. Is that the intended default, or is it worth clarifying (for example, that there is no default beyond it being non-public information)?